internal/restorer/restorer.go

```diff
@@ -305,6 +307,59 @@ func (res *Restorer) RestoreTo(ctx context.Context, dst string) error {
 	return err
 }
 
+// verifyFile checks if the given target file's contents match the
+// blobs given in node
+// It first checks the file size and then checks all file blobs
+// It always returns the successfully checked blobs
+func (res *Restorer) verifyFile(target string, node *restic.Node) (matchingBlobs []bool, err error) {
+	debug.Log("calling verifyFile")
+
+	matchingBlobs = make([]bool, len(node.Content))
+	stat, err := os.Stat(target)
+	if err != nil {
+		return
+	}
+	if int64(node.Size) != stat.Size() {
+		return matchingBlobs, errors.Errorf("Invalid file size: expected %d got %d", node.Size, stat.Size())
+	}
+
+	file, err := os.Open(target)
+	if err != nil {
+		return
+	}
+
+	offset := int64(0)
+	wrongOffset := offset
+	foundNotMatch := false
+	for i, blobID := range node.Content {
+		length, _ := res.repo.LookupBlobSize(blobID, restic.DataBlob)
+		buf := make([]byte, length) // TODO do I want to reuse the buffer somehow?
```

Comment:
Instead of reading the whole blob into memory to get its hash, you could use a SectionReader with a smaller buffer:

```go
sha := sha256.New()
buf := make([]byte, 8192)
for i, blobID := range node.Content {
	length, _ = res.repo.LookupBlobSize(blobID, restic.DataBlob)
	block := io.NewSectionReader(file, offset, int64(length))

	h.Reset()
	_, err = io.CopyBuffer(sha, block, buf)
	// check error

	h := sha.Sum(buf)
	// ...
}
```

Reply:
As the size of a blob is limited to a few MB, it shouldn't hurt to load the full blob?

Reply:
I just copied the already exiting code from VerifyFiles into this new method VerifyFile. I would not change this in this PR.